src/web/js/beforePyret.js

```diff
@@ -48,15 +48,127 @@ window.stickMessage = function(message) {
   var err = $("<div>").addClass("active").text(message);
   $(".notificationArea").prepend(err);
 };
+window.mkWarningUpper = function(){return $("<div class='warning-upper'>");}
+window.mkWarningLower = function(){return $("<div class='warning-lower'>");}
 
 $(window).bind("beforeunload", function() {
   return "Because this page can load slowly, and you may have outstanding changes, we ask that you confirm before leaving the editor in case closing was an accident.";
 });
+
+CodeMirror.defineDocExtension("rangeHandle", function(position, options) {
+  var inclusiveLeft,
+      inclusiveRight,
+      shared;
+  if (options === undefined) {
+    inclusiveLeft = false,
+    inclusiveRight = false;
+    shared = true;
+  } else {
+    if (options.inclusiveLeft !== undefined)
+      inclusiveLeft = options.inclusiveLeft;
+    if (options.inclusiveRight !== undefined)
+      inclusiveRight = options.inclusiveRight;
+    if (options.shared !== undefined)
+      shared = options.shared;
+  }
+  return this.markText(position.from, to,
+    { inclusiveLeft   : inclusiveLeft,
+      inclusiveRight  : inclusiveRight,
+      shared          : shared  });
+});
+
```

Comment:
Lines 80-86 set `Position` to `undefined`, right?  Unless there's something special about this context that I'm not getting; I get `undefined` for `Position` if I use this at the console.

Is this dead code, and some other definition of `Position` is actually being used?  There seem to be a bunch of things named `Position`.